test/common/upstream/health_checker_impl_test.cc

```diff
@@ -96,6 +96,13 @@ class TestHttpHealthCheckerImpl : public HttpHealthCheckerImpl {
   MOCK_METHOD1(createCodecClient_, Http::CodecClient*(Upstream::Host::CreateConnectionData&));
 
   Http::CodecClient::Type codecClientType() { return codec_client_type_; }
+
+  void closeConnections() {
+    for (auto& session : active_sessions_) {
+      dynamic_cast<HttpActiveHealthCheckSession*>(session.second.get())
```

Comment:
this is pretty gross, but it served to reproduced the crash in the reported issue. would love suggestions on how to do this in a cleaner way

Reply:
Does calling close on the `client_connection_` instances held under `test_sessions_` not work?

You might need to let the dispatcher run, but I think that's a better simulation of a closed connection?

Reply:
Wasn't aware of that, I'll give that a go